DocCRender.docc/contributing/Internals.md

```diff
@@ -0,0 +1,189 @@
+* The `core` styles are used to define Sass variables, mixins and functions, that are to be used to generate reusable and complex styles inside Vue files. 
+
+> Important: SCSS files inside **core** should never output css directly! These style files are imported in almost every vue file, and making them output styles, will lead to duplication of styles across the entire app. 
+
+If you need access to a global mixin or var, from `_core.scss` you have to import it first.
+
+```html
+<style scoped lang="scss">
+    @import '@/styles/_core.scss';
+    .some-class {
+        margin-left: $varfrom-core-folder;
+    }
+</style>
+```
+
+## Unit tests
+
+DocC Render is unit tested, using the [Jest](https://jestjs.io/) testing framework. All components and utilities strive to have a corresponding test file inside the  `tests` folder, mirroring the file's location.
+
+Unit tests can be run via:
+
+* `npm run test:unit`  - run all unittests
+* `npm run test:unit:watch`  - run only the modified files and watch for changes
+* `npx run test:unit tests/unit/path/to/spec.js`  - run an individual test file
+
+## Code style
+
+DocC Render uses [ESLint](https://eslint.org/) to format the code and avoid syntax errors. It is following [Airbnb](https://github.com/airbnb/javascript) code styling along with [Vue Style Guide](https://vuejs.org/v2/style-guide/)’s essential configuration.
+
+To run the linter run:
+
+`npm run lint`
+
+To automatically fix any error, run:
+
+`npm run lint:fix`
+
+If you use the development server via `npm run serve`, files are linted on each change. Read more about configuring your linter [here](https://cli.vuejs.org/core-plugins/eslint.html#configuration)
+
+## Environment flags
+
+DocC Render has a few build-time environment flags, that allow you to set configuration parameters before building. Read [How to set Environment Variables in Vue CLI](https://cli.vuejs.org/guide/mode-and-env.html#environment-variables).
+
+**List of env variables:**
+
+* **VUE_APP_DEV_SERVER_PROXY** - The endpoint or path to read render JSON from
+* **VUE_APP_TITLE** - The default page title to apply to pages
+
+## Available Scripts
+
+These are the most commonly used npm scripts you would need to develop DocC Render:
+
+* **serve** - used while developing DocC Render or a theme. Starts up a custom live-reloading server, which serves a local DocC Render instance.
+* **build** - builds DocC Render, in a state ready for `docc`.
+* **test** - run the entire DocC Render test suite.
+* **lint** - run the linter, to check for code styling errors.
+
+### Configuring the server
+
+DocC Render uses browser APIs to simulate nested file structures, which means we need to tell the server to redirect those nested paths properly. To see examples for popular server configs, check: [Vue Router Server Configs](https://router.vuejs.org/guide/essentials/history-mode.html#example-server-configurations) 
```

Comment:
The wording of this paragraph is a little confusing to me, but I don't have a better suggestion off the top of my head right now. I'll try to circle back on this one.